lib/plugins/aws/package/compile/events/httpApi/index.js

```diff
@@ -48,9 +48,27 @@ class HttpApiEvents {
       },
     };
 
-    // TODO: Complete schema, see https://github.com/serverless/serverless/issues/8021
     this.serverless.configSchemaHandler.defineFunctionEvent('aws', 'httpApi', {
-      anyOf: [{ type: 'string' }, { type: 'object' }],
+      anyOf: [
+        { type: 'string' },
```

Comment:
It'll be nice to make it more restrictive with regexp pattern.

I guess we can improve regexp at `methodPathPattern` so it's:

```javascript
const methodPathPattern = new RegExp(
  `^(\\*|(?:${Array.from(allowedMethods).join('|')}))( .+)?$`,
  'i'
);
```

And define property with [`regexp`](https://github.com/ajv-validator/ajv-keywords#regexp) setting

Also having that, it'll be nice to simplify validation logic in `resolveConfiguration`, as over there we may blindly assume that format is respected. So we should cover only cases which are not handled by schema

Reply:
@medikoo, noted, will update accordingly :)
One last thing on that matter : some error messages lose in clarity
Suppose you define for a function `hello` the `httpApi` attribute equals to the string `myPath`, the following error messages are displayed :
- before schema validation : `Invalid "<method> <path>" route in function hello for httpApi event in serverless.yml`
- after schema validation : `Configuration error at 'functions.hello.events[0].httpApi': should pass "regexp" keyword validation`

One way to compensate would be to define custom [JSON Schema error messages](https://github.com/ajv-validator/ajv-errors#messages-for-keywords) based on validation keywords. WDYT ?

Reply:
@fredericbarthelet that's very good point.

In light of that I think we should improve messaging of `regex` keyword errors. It easily can be done here: https://github.com/serverless/serverless/blob/20d9c6414af9a06e2479d203e62aa6427a80f87f/lib/classes/ConfigSchemaHandler/normalizeAjvErrors.js#L210-L233

Let's replace the message, with e.g. `doesn't match expected format` (if you feel you can come up with something better, please propose :)

Reply:
@medikoo, I added a custom message for `regexp` keyword : `value '${error.data}' does not satisfy pattern ${error.schema}`. The resulting error message in the scenario described above, now much clearer, is :

> Configuration error at 'functions.hello.events[0].httpApi': value "myPath" does not satisfy pattern /^(?:\*|(GET|POST|PUT|PATCH|OPTIONS|HEAD|DELETE) (\/\S*))$/i

WDYT ?

Reply:
Looks good. I wonder whether listing pattern, won't be too noisy, but it's definitely very informative, so I'd say it's great!